lnrpc/routerrpc/router_backend.go

```diff
@@ -504,9 +536,11 @@ func (r *RouterBackend) extractIntentFromSendRequest(
 		return nil, errors.New("timeout_seconds must be specified")
 	}
 
-	var destTLV map[uint64][]byte
-	if len(destTLV) != 0 {
-		payIntent.FinalDestRecords = tlv.MapToRecords(destTLV)
+	payIntent.FinalDestRecords, err = UnmarshallCustomRecords(
+		rpcPayReq.DestTlv,
+	)
+	if err != nil {
+		return payIntent, err
```

Comment:
nit: returned value is ignored anyway, could also just be `nil, err`.

Reply:
fixed